gtkwave.nsi

```diff
@@ -0,0 +1,179 @@
+  File ${SYSROOT}\bin\libcairo-2.dll
+  File ${SYSROOT}\bin\libcairo-gobject-2.dll
+  File ${SYSROOT}\bin\zlib1.dll
+  File ${SYSROOT}\bin\libglib-2.0-0.dll
+  File ${SYSROOT}\bin\libatk-1.0-0.dll
+  File ${SYSROOT}\bin\libgobject-2.0-0.dll
+  File ${SYSROOT}\bin\libgmodule-2.0-0.dll
+  File ${SYSROOT}\bin\libgthread-2.0-0.dll
+  File ${SYSROOT}\bin\libpango-1.0-0.dll
+  File ${SYSROOT}\bin\libfribidi-0.dll
+  File ${SYSROOT}\bin\libpangocairo-1.0-0.dll
+  File ${SYSROOT}\bin\libpangoft2-1.0-0.dll
+  File ${SYSROOT}\bin\libpangowin32-1.0-0.dll
+  File ${SYSROOT}\bin\libpng16-16.dll
+  File ${SYSROOT}\bin\libtiff-5.dll
+  File ${SYSROOT}\bin\libjpeg-62.dll
+  File ${SYSROOT}\bin\libfontconfig-1.dll
+  File ${SYSROOT}\bin\libxml2-2.dll
+  File ${SYSROOT}\bin\libfreetype-6.dll
+  File ${SYSROOT}\bin\gdk-pixbuf-query-loaders.exe
+  File ${SYSROOT}\bin\${LIBGCCDLL}
+  File ${SYSROOT}\bin\libstdc++-6.dll
+  File ${SYSROOT}\bin\libwinpthread-1.dll
+  File ${SYSROOT}\bin\libgfortran-*.dll
+  File ${SYSROOT}\bin\librsvg-2-2.dll
+  File ${SYSROOT}\bin\libcroco-0.6-3.dll
+  File ${SYSROOT}\bin\libxml2-2.dll
+  File ${SYSROOT}\bin\libquadmath-0.dll
+  File ${SYSROOT}\bin\libopenjp2.dll
+  File ${SYSROOT}\bin\libsqlite3-0.dll
+  File ${SYSROOT}\bin\libexpat-1.dll
+  File ${SYSROOT}\bin\libbz2-1.dll
+  #File ${SYSROOT}\bin\libgdkglext-win32-1.0-0.dll
+  #File ${SYSROOT}\bin\libgtkglext-win32-1.0-0.dll
+  File ${SYSROOT}\bin\glew32.dll
+  File ${SYSROOT}\bin\libzip-5.dll
+  File ${SYSROOT}\bin\libharfbuzz-0.dll
+  File ${SYSROOT}\bin\libharfbuzz-icu-0.dll
+  File ${SYSROOT}\bin\libharfbuzz-subset-0.dll
+  File ${SYSROOT}\bin\gspawn-win64-helper.exe
+  File ${SYSROOT}\bin\gdbus.exe
+  File ${SYSROOT}\bin\libsigc-2.0-0.dll
+  File ${SYSROOT}\bin\libmpfr-6.dll
+  File ${SYSROOT}\bin\libmpfr-6.dll
+  # Libepoxy in official package is broken, so we use an older working version!
+  # File ${SYSROOT}\bin\libepoxy-0.dll
+  File \usr\local\mingw64\bin\libepoxy-0.dll
+  File \usr\local\mingw64\bin\libfmt.dll
+  File \usr\local\mingw64\bin\libgmp-10.dll
+  File ${OUTDIR}\src\${NAME}.exe
+  
+  
+  SetOutPath "$INSTDIR"
+  File /r ${SYSROOT}\etc
+  SetOutPath $INSTDIR\lib\gdk-pixbuf-2.0\2.10.0\loaders
+  File ${SYSROOT}\lib\gdk-pixbuf-2.0\2.10.0\loaders\*
+  SetOutPath $INSTDIR\lib\gdk-pixbuf-2.0\2.10.0
+  File ${SYSROOT}\lib\gdk-pixbuf-2.0\2.10.0\loaders.cache
+  #SetOutPath $INSTDIR\lib\gtk-2.0\2.10.0\engines
+  #File ${SYSROOT}\lib\gtk-2.0\2.10.0\engines\*
+  
+  SetOutPath $INSTDIR\share
+  File /r ${SYSROOT}\share\themes
+  SetOutPath $INSTDIR\share\icons
+  File /r ${SYSROOT}\share\icons\Adwaita
+  
+  SetOutPath $INSTDIR\share\glib-2.0
+  File /r ${SYSROOT}\share\glib-2.0\schemas
+  
+  # Build the gdk-pixbuf.loaders file automatically
+  ExpandEnvStrings $0 %COMSPEC%
+  nsExec::ExecToStack '"$0" /C ""$INSTDIR\bin\gdk-pixbuf-query-loaders" > "$INSTDIR\lib\gdk-pixbuf-2.0\2.10.0\loaders.cache""'
+  
+  ; Set up association with .vcd files
```

Comment:
What about the other filetypes? At least `.gtkw` and `.fst` should also be associated, in my opinion.

Reply:
I removed this, since it has nothing to do with the windows complation.

However there is still a issue with the paiinting. Part of the problem is that you are drawing through the pattern (where M=MoveTo L=LineTo S=Stroke) MLSMLSMLS... instead of MLLLLLL...S which is the prefered way of using cairo.

Reply:
> However there is still a issue with the paiinting.

What kind of issue, performance or visual artefacts?

> Part of the problem is that you are drawing through the pattern (where M=MoveTo L=LineTo S=Stroke) MLSMLSMLS... instead of MLLLLLL...S which is the prefered way of using cairo.

The current drawing code is very old and hasn't changed a lot since the GTK 1 days. I've done some extensive refactoring of the drawing code a while back, which I originally intended to merge in as part of the GTK 4 port, but I'm considering to merge it in earlier for GTKWave 3.4. The new code should make it easier to optimize the calls that are made to the cairo API.

I had done some tests with batching of strokes as part of #224, but didn't find any significant performance improvement when I try to batch a lot of strokes. Around ~10 lines per stroke seemed to give the best performance. There was some performance anomaly if I tried to draw thousands of lines per stroke, but I didn't investigate that further.

The `MLLLLL...S` pattern is often not possible, because the line color changes between segments. `MLML...S` is the best we could do in these situations.

Reply:
I was talking about the rendering aesthetics, and not about speed. E.g. the stereo.vcd file (which I found in some discussion list) which I found on some discussion list. I investigated the rendering problems today and found that there are to reasons for the bad rendering.

1. The main reason is that the coordinates are turned into integers before they are passed into `XXX_gdk_draw_line`. This causes lots of x-coordinate  collisions (because multiple time points are rounded to the same x-coordinates). This is rounding interfers with cairo's ability to calculate the pixel coverages.
2. The fact that you stroke after each ML. I agree that doing MLMLMLML...S would be ok. But doing MLSMLS.... causes multiple renderings of the same pixels, and the line become thicker than intended.

I working on an experimental branch where I changed the integer coordinates to double and solved these issues quite hackishly. I hope to upload a new PR with the changes, but if you are working on a GTK4 branch then it might be redundant.

But this has nothing to do with the Windows and the installers. :-) 

[stereo.zip](https://github.com/gtkwave/gtkwave/files/12409788/stereo.zip)

Reply:
> But this has nothing to do with the Windows and the installers. :-)

Yes, let's continue the discussion in the issue you've opened.